pkg/resources/templates/config/nifi_properties.go

```diff
@@ -183,31 +239,67 @@ nifi.security.user.oidc.read.timeout=5 secs
 nifi.security.user.oidc.preferred.jwsalgorithm=
+nifi.security.user.oidc.additional.scopes=
+nifi.security.user.oidc.claim.identifying.user=
+nifi.security.user.oidc.fallback.claims.identifying.user=
+nifi.security.user.oidc.claim.groups=groups
+nifi.security.user.oidc.truststore.strategy=JDK
+nifi.security.user.oidc.token.refresh.window=60 secs
 
 # Apache Knox SSO Properties #
 nifi.security.user.knox.url=
 nifi.security.user.knox.publicKey=
 nifi.security.user.knox.cookieName=hadoop-jwt
 nifi.security.user.knox.audiences=
 
+# SAML Properties #
+nifi.security.user.saml.idp.metadata.url=
+nifi.security.user.saml.sp.entity.id=
+nifi.security.user.saml.identity.attribute.name=
+nifi.security.user.saml.group.attribute.name=
+nifi.security.user.saml.request.signing.enabled=false
+nifi.security.user.saml.want.assertions.signed=true
+nifi.security.user.saml.signature.algorithm=http://www.w3.org/2001/04/xmldsig-more#rsa-sha256
+nifi.security.user.saml.authentication.expiration=12 hours
+nifi.security.user.saml.single.logout.enabled=false
+nifi.security.user.saml.http.client.truststore.strategy=JDK
+nifi.security.user.saml.http.client.connect.timeout=30 secs
+nifi.security.user.saml.http.client.read.timeout=30 secs
+
 # Identity Mapping Properties #
 # These properties allow normalizing user identities such that identities coming from different identity providers
 # (certificates, LDAP, Kerberos) can be treated the same internally in NiFi. The following example demonstrates normalizing
 # DNs from certificates and principals from Kerberos into a common identity string:
 #
 # nifi.security.identity.mapping.pattern.dn=^CN=(.*?), OU=(.*?), O=(.*?), L=(.*?), ST=(.*?), C=(.*?)$
 # nifi.security.identity.mapping.value.dn=$1@$2
+# nifi.security.identity.mapping.transform.dn=NONE
 # nifi.security.identity.mapping.pattern.kerb=^(.*?)/instance@(.*?)$
 # nifi.security.identity.mapping.value.kerb=$1@$2
+# nifi.security.identity.mapping.transform.kerb=UPPER
+
+# Group Mapping Properties #
+# These properties allow normalizing group names coming from external sources like LDAP. The following example
+# lowercases any group name.
+#
+# nifi.security.group.mapping.pattern.anygroup=^(.*)$
+# nifi.security.group.mapping.value.anygroup=$1
+# nifi.security.group.mapping.transform.anygroup=LOWER
+
+# Listener Bootstrap properties #
+# This property defines the port used to listen for communications from NiFi Bootstrap. If this property
+# is missing, empty, or 0, a random ephemeral port is used.
+nifi.listener.bootstrap.port=0
 
 # cluster common properties (all nodes must have same values) #
 nifi.cluster.protocol.heartbeat.interval=5 sec
+nifi.cluster.protocol.heartbeat.missable.max=8
 nifi.cluster.protocol.is.secure={{ .ClusterSecure }}
 
 # cluster node properties (only configure for cluster nodes) #
-
 nifi.cluster.is.node={{.IsNode}}
 nifi.cluster.node.protocol.threads=10
+nifi.cluster.leader.election.implementation=CuratorLeaderElectionManager
```

Comment:
```suggestion
# TODO: implement in operator the Kubernetes integration with configmap
nifi.cluster.leader.election.implementation=CuratorLeaderElectionManager
```

Reply:
I'm most excited to test with the ConfigMap provider so we can drop ZooKeeper all together.

Reply:
Yeah our docs assume you need a zookeeper, so explaining how to configure zookeeper or kubernetes would be a good idea. There's also the topic of migrating from zookeeper to kubernetes, but that is something we can address in a later PR. I don't believe it's fully implemented in NiFi, yet.

Is there an API endpoint that would provide the name of the configmap? If so, nifikop could use that

Reply:
For the ConfigMap, the implementation in NiFi clearly assumes only one NiFi cluster per namespace. Here's how NiFi finds the components storing state in ConfigMaps: https://github.com/apache/nifi/blob/main/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-kubernetes-bundle/nifi-framework-kubernetes-state-provider/src/main/java/org/apache/nifi/kubernetes/state/provider/KubernetesConfigMapStateProvider.java#L329-L338

So this would be a new constraint users of this feature should be aware of unless the state management implementation adds a label or something to the ConfigMaps to identify them by the cluster they belong to. Honestly, i think this should be contributed to NiFi because it's an unnecessary constraint.

Reply:
There is not 1 ConfigMap but several of them. It's 1 ConfigMap for 1 Component. The name of them are `nifi-component-%s` with `%s`, the id of the processor.

Reply:
Right, that function I linked finds all component IDs (one per config map) within the namespace they get created in so it assumes there is only one NiFi cluster in the namespace. That is, there's no filter for "this NiFi cluster", for example.

IMO, NiFi should allow for multiple clusters per namespace so I think NiFi's implementation should change.

Reply:
I started a conversation on the Slack, if you want to participate: https://apachenifi.slack.com/archives/C0L9VCD47/p1704727400482319

Reply:
Yes, I will try to do the documentation by the end of the week. And in it I will indicate the current limitation.